src/print.jl

```diff
@@ -162,22 +163,25 @@ function Base.show(io::IO, model::Model)
     println(io, "Variable", plural(num_variables(model)), ": ",
             num_variables(model))
     if sense != MOI.FEASIBILITY_SENSE
-        if model.nlp_data !== nothing && model.nlp_data.nlobj !== nothing
-            println(io, "Objective function type: Nonlinear")
-        else
+        nlobj = nlp_objective_function(model)
```

Comment:
So `AbstractModel`s have to implement `nlp_objective_function`? That's weird. I'm tempted to say that we shouldn't implement `show` for an `AbstractModel`.

Reply:
I have added `show_objective_function_summary` an `AbstractModel` should implement that instead